newIDE/app/src/MainFrame/index.js

```diff
@@ -307,60 +352,91 @@ class MainFrame extends React.Component<Props, State> {
       });
   };
 
-  loadFromSerializedProject = (
+  const loadFromSerializedProject = (
     serializedProject: gdSerializerElement,
-    fileMetadata: ?FileMetadata
-  ): Promise<void> => {
+    fileMetadata: ?FileMetadata,
+    newState: {
+      currentProject: ?gdProject,
+      editorTabs: EditorTabsState,
+    },
+    eventsFunctionsExtensionsState = props.eventsFunctionsExtensionsState
+  ): Promise<State> => {
     return timePromise(
       () => {
         const newProject = gd.ProjectHelper.createNewGDJSProject();
         newProject.unserializeFrom(serializedProject);
-
-        return this.loadFromProject(newProject, fileMetadata);
+        return loadFromProject(
+          newProject,
+          fileMetadata,
+          newState,
+          eventsFunctionsExtensionsState
+        ).then(state => Promise.resolve(state));
```

Comment:
post-nitpicking: you don't need to resolve when you're already in a promise. Also this then is not necessary ;)

Reply:
Should I open a new PR to change this?

Reply:
No I'm on it :)